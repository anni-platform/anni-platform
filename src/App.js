import React, { Component } from 'react';
import { createStore, compose } from 'redux';
import Navigation from 'components/Navigation';
import { Provider } from 'react-redux';
import reducer from 'reducers';
import { getAccessTokenFromUrl, storeSessionToken, getAuthUrl } from 'adapters';
import { saveState } from './utils/localStorage';
import StaticJSONFileDatabase from 'utils/fileStorage';
import { Loader, Wrapper } from 'styled';
import GlobalStyles from 'styled/components/Base';

let enhancer = null;
let DevTools = null;
if (process.env.NODE_ENV === 'development') {
  DevTools = require('DevTools').default;
  enhancer = compose(DevTools.instrument());
}

const test = {
  a: 1,
  b: 2,
};

class App extends Component {
  constructor() {
    super();
    this.state = { store: null };
  }
  componentDidMount() {
    const token = getAccessTokenFromUrl();
    window.loginDropbox = function() {
      window.location.href = getAuthUrl();
    };
    if (token) {
      storeSessionToken(token);
    }
    StaticJSONFileDatabase.hydrateStoreFromFileDatabase().then(data => {
      let store;
      if (process.env.NODE_ENV === 'development') {
        store = createStore(reducer, data, enhancer);
      } else {
        store = createStore(reducer, data);
      }
      store.subscribe(() => {
        saveState(store.getState());
      });
      this.setState({ store });
    });
  }
  render() {
    GlobalStyles();

    const store = this.state.store;
    return store ? (
      <Provider store={store}>
        <Wrapper>
          <Navigation />
          {this.props.children}
          {DevTools ? <DevTools /> : null}
        </Wrapper>
      </Provider>
    ) : (
      <Loader fullPage />
    );
  }
}

export default App;
