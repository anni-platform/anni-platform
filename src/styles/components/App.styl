.Loading {
    padding: 1rem;
}