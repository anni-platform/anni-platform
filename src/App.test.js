import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
