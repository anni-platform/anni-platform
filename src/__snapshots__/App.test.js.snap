exports[`test renders without crashing 1`] = `
<div
  className="Loading">
  Loading...
</div>
`;
